patroni/dcs/kubernetes.py

```diff
@@ -910,55 +932,62 @@ def _cluster_from_nodes(self, group, nodes, pods):
+                clusters[group]['pods'][name] = pod
 
         for name, kind in path['nodes'].items():
             group = kind.metadata.labels.get(self._CITUS_LABEL)
             if group and citus_group_re.match(group):
                 clusters[group]['nodes'][name] = kind
-        return {int(group): self._cluster_from_nodes(group, value['nodes'], value['pods'])
+        return {int(group): self._cluster_from_nodes(group, value['nodes'], value['pods'].values())
                 for group, value in clusters.items()}
 
-    def __load_cluster(self, group, loader):
+    def __load_cluster(
+            self, group: Optional[str], loader: Callable[[Dict[str, Any]], Union[Cluster, Dict[int, Cluster]]]
+    ) -> Union[Cluster, Dict[int, Cluster]]:
+        assert self._retry.deadline is not None
         stop_time = time.time() + self._retry.deadline
         self._api.refresh_api_servers_cache()
         try:
             with self._condition:
                 self._wait_caches(stop_time)
-                pods = [pod for pod in self._pods.copy().values()
-                        if not group or pod.metadata.labels.get(self._CITUS_LABEL) == group]
+                pods = {name: pod for name, pod in self._pods.copy().items()
+                        if not group or pod.metadata.labels.get(self._CITUS_LABEL) == group}
                 nodes = {name: kind for name, kind in self._kinds.copy().items()
                          if not group or kind.metadata.labels.get(self._CITUS_LABEL) == group}
             return loader({'group': group, 'pods': pods, 'nodes': nodes})
         except Exception:
             logger.exception('get_cluster')
             raise KubernetesError('Kubernetes API is not responding properly')
 
-    def _load_cluster(self, path, loader):
+    def _load_cluster(
+            self, path: str, loader: Callable[[Any], Union[Cluster, Dict[int, Cluster]]]
+    ) -> Union[Cluster, Dict[int, Cluster]]:
         group = self._citus_group if path == self.client_path('') else None
         return self.__load_cluster(group, loader)
 
-    def get_citus_coordinator(self):
+    def get_citus_coordinator(self) -> Optional[Cluster]:
         try:
-            return self.__load_cluster(str(CITUS_COORDINATOR_GROUP_ID), self._cluster_loader)
+            ret = self.__load_cluster(str(CITUS_COORDINATOR_GROUP_ID), self._cluster_loader)
+            assert isinstance(ret, Cluster)
+            return ret
         except Exception as e:
             logger.error('Failed to load Citus coordinator cluster from Kubernetes: %r', e)
 
     @staticmethod
-    def compare_ports(p1, p2):
+    def compare_ports(p1: K8sObject, p2: K8sObject) -> bool:
         return p1.name == p2.name and p1.port == p2.port and (p1.protocol or 'TCP') == (p2.protocol or 'TCP')
 
     @staticmethod
-    def subsets_changed(last_observed_subsets, ip, ports):
+    def subsets_changed(last_observed_subsets: List[K8sObject], ip: str, ports: List[K8sObject]) -> bool:
```

Comment:
To match docstring example
```suggestion
    def subsets_changed(last_observed_subsets: List[K8sObject], ip: Optional[str], ports: List[K8sObject]) -> bool:
```

Reply:
It could never happen on practice...